lib/diff.ml

```diff
@@ -110,6 +147,21 @@ and module_declaration ~typing_env ~name ~reference ~current =
       modtype_item ~loc:reference.md_loc ~typing_env ~name
         ~reference:ref_modtype ~current:curr_modtype
 
+and class_declaration ~typing_env ~name ~reference ~current = 
+  match (reference.cls_decl_type, current.cls_decl_type) with
+  | Cty_signature ref_sig, Cty_signature curr_sig -> 
+      class_signatures ~typing_env ~reference:ref_sig ~current:curr_sig
+      |> Option.map (fun cdiff -> Class { cname = name; cdiff })
+  | ref_class_type, curr_class_type -> 
+      class_type_item ~loc:reference.cls_decl_loc ~typing_env ~name
+        ~reference:ref_class_type ~current:curr_class_type 
+and class_type_item ~loc ~typing_env ~name ~reference ~current =
+          (* Logic to compare reference and current class_type *)
```

Comment:
Until now, do not try to compare class/class types. The issue is only about detecting addition and removal.

Reply:
OK, thank you sir. I will proceed with this.

Reply:
Hello sir, i made some changes could you have a look please?